### PR TITLE
Clean up and bump dependencies

### PR DESCRIPTION
**Pull request by friedrichwilken**
**Description**

For improving performance, security, and stability this PR proposes the following changes:

- substitute the deprecated `io/ioutil` with `io` and `os`
- minor code cleanup e. g. simplifying if-return statements
- bump `k8s.io/api` and and `k8s.io/client-go` to `v0.28.5`
- bump `github.com/onsi/gomega` to `v1.29.0`

### CODE REVIEW

File: components/kyma-metrics-collector/pkg/process/event_stream.go [hunk 1]
```diff
@@ -43,13 +41,7 @@ type Input struct {
 	svcList  *corev1.ServiceList
 }
 
-type NodeInfo struct {
-	cpu    int
-	memory int
-}
```
Comment (mfaizanse) [inline components/kyma-metrics-collector/pkg/process/event_stream.go:49]: Why is this removed?

Comment (friedrichwilken) [reply components/kyma-metrics-collector/pkg/process/event_stream.go:49]: It is not used.

**Review by mfaizanse — APPROVED** [review]
